// cezerin2-client/lib/api/theme/placeholders.js
"use strict";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

Object.defineProperty(exports, "__esModule", {
  value: true
});

var ThemePlaceholders = /*#__PURE__*/function () {
  function ThemePlaceholders(client) {
    _classCallCheck(this, ThemePlaceholders);

    this.client = client;
    this.resourceUrl = "/theme/placeholders";
  }

  _createClass(ThemePlaceholders, [{
    key: "list",
    value: function list() {
      return this.client.get(this.resourceUrl);
    }
  }, {
    key: "retrieve",
    value: function retrieve(placeholderKey) {
      return this.client.get("".concat(this.resourceUrl, "/").concat(placeholderKey));
    }
  }, {
    key: "create",
    value: function create(data) {
      return this.client.post(this.resourceUrl, data);
    }
  }, {
    key: "update",
    value: function update(placeholderKey, data) {
      return this.client.put("".concat(this.resourceUrl, "/").concat(placeholderKey), data);
    }
  }, {
    key: "delete",
    value: function _delete(placeholderKey) {
      return this.client.delete("".concat(this.resourceUrl, "/").concat(placeholderKey));
    }
  }]);

  return ThemePlaceholders;
}();

exports.default = ThemePlaceholders;
