// cezerin2-client/lib/webstore/services.js
"use strict";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

Object.defineProperty(exports, "__esModule", {
  value: true
});

var WebStoreServices = /*#__PURE__*/function () {
  function WebStoreServices(client) {
    _classCallCheck(this, WebStoreServices);

    this.client = client;
    this.resourceUrl = "/services";
  }

  _createClass(WebStoreServices, [{
    key: "list",
    value: function list(filter) {
      return this.client.get(this.resourceUrl, filter);
    }
  }, {
    key: "retrieve",
    value: function retrieve(id) {
      return this.client.get("".concat(this.resourceUrl, "/").concat(id));
    }
  }, {
    key: "enable",
    value: function enable(id) {
      return this.client.post("".concat(this.resourceUrl, "/").concat(id, "/enable"));
    }
  }, {
    key: "disable",
    value: function disable(id) {
      return this.client.post("".concat(this.resourceUrl, "/").concat(id, "/disable"));
    }
  }]);

  return WebStoreServices;
}();

exports.default = WebStoreServices;
