// cezerin2-client/lib/api/checkoutFields.js
"use strict";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

Object.defineProperty(exports, "__esModule", {
  value: true
});

var CheckoutFields = /*#__PURE__*/function () {
  function CheckoutFields(client) {
    _classCallCheck(this, CheckoutFields);

    this.client = client;
    this.resourceUrl = "/settings/checkout/fields";
  }

  _createClass(CheckoutFields, [{
    key: "list",
    value: function list() {
      return this.client.get(this.resourceUrl);
    }
  }, {
    key: "retrieve",
    value: function retrieve(name) {
      return this.client.get("".concat(this.resourceUrl, "/").concat(name));
    }
  }, {
    key: "update",
    value: function update(name, data) {
      return this.client.put("".concat(this.resourceUrl, "/").concat(name), data);
    }
  }]);

  return CheckoutFields;
}();

exports.default = CheckoutFields;
