// cezerin2-client/lib/api/customers.js
"use strict";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

Object.defineProperty(exports, "__esModule", {
  value: true
});

var Customers = /*#__PURE__*/function () {
  function Customers(client) {
    _classCallCheck(this, Customers);

    this.client = client;
    this.resourceUrl = "/customers";
  }

  _createClass(Customers, [{
    key: "list",
    value: function list(filter) {
      return this.client.get(this.resourceUrl, filter);
    }
  }, {
    key: "retrieve",
    value: function retrieve(id, filter) {
      return this.client.get("".concat(this.resourceUrl, "/").concat(id), filter);
    }
  }, {
    key: "create",
    value: function create(data) {
      return this.client.post(this.resourceUrl, data);
    }
  }, {
    key: "update",
    value: function update(id, data) {
      return this.client.put("".concat(this.resourceUrl, "/").concat(id), data);
    }
  }, {
    key: "delete",
    value: function _delete(id) {
      return this.client.delete("".concat(this.resourceUrl, "/").concat(id));
    }
  }, {
    key: "createAddress",
    value: function createAddress(customerId, data) {
      return this.client.post("".concat(this.resourceUrl, "/").concat(customerId), data);
    }
  }, {
    key: "updateAddress",
    value: function updateAddress(customerId, addressId, data) {
      return this.client.put("".concat(this.resourceUrl, "/").concat(customerId, "/addresses/").concat(addressId), data);
    }
  }, {
    key: "deleteAddress",
    value: function deleteAddress(customerId, addressId) {
      return this.client.delete("".concat(this.resourceUrl, "/").concat(customerId, "/addresses/").concat(addressId));
    }
  }, {
    key: "setDefaultBillingAddress",
    value: function setDefaultBillingAddress(customerId, addressId) {
      return this.client.post("".concat(this.resourceUrl, "/").concat(customerId, "/addresses/").concat(addressId, "/default_billing"));
    }
  }, {
    key: "setDefaultShippingAddress",
    value: function setDefaultShippingAddress(customerId, addressId) {
      return this.client.post("".concat(this.resourceUrl, "/").concat(customerId, "/addresses/").concat(addressId, "/default_shipping"));
    }
  }]);

  return Customers;
}();

exports.default = Customers;
