// cezerin2-client/lib/api/productCategories.js
"use strict";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

Object.defineProperty(exports, "__esModule", {
  value: true
});

var ProductCategories = /*#__PURE__*/function () {
  function ProductCategories(client) {
    _classCallCheck(this, ProductCategories);

    this.client = client;
    this.resourceUrl = "/product_categories";
  }

  _createClass(ProductCategories, [{
    key: "list",
    value: function list(filter) {
      return this.client.get(this.resourceUrl, filter);
    }
  }, {
    key: "retrieve",
    value: function retrieve(id) {
      return this.client.get("".concat(this.resourceUrl, "/").concat(id));
    }
  }, {
    key: "create",
    value: function create(data) {
      return this.client.post(this.resourceUrl, data);
    }
  }, {
    key: "update",
    value: function update(id, data) {
      return this.client.put("".concat(this.resourceUrl, "/").concat(id), data);
    }
  }, {
    key: "delete",
    value: function _delete(id) {
      return this.client.delete("".concat(this.resourceUrl, "/").concat(id));
    }
  }, {
    key: "uploadImage",
    value: function uploadImage(categoryId, formData) {
      return this.client.postFormData("".concat(this.resourceUrl, "/").concat(categoryId, "/image"), formData);
    }
  }, {
    key: "deleteImage",
    value: function deleteImage(id) {
      return this.client.delete("".concat(this.resourceUrl, "/").concat(id, "/image"));
    }
  }]);

  return ProductCategories;
}();

exports.default = ProductCategories;
