// cezerin2-client/lib/api/products/options.js
"use strict";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

Object.defineProperty(exports, "__esModule", {
  value: true
});

var ProductOptions = /*#__PURE__*/function () {
  function ProductOptions(client) {
    _classCallCheck(this, ProductOptions);

    this.client = client;
  }

  _createClass(ProductOptions, [{
    key: "list",
    value: function list(productId) {
      return this.client.get("/products/".concat(productId, "/options"));
    }
  }, {
    key: "retrieve",
    value: function retrieve(productId, optionId) {
      return this.client.get("/products/".concat(productId, "/options/").concat(optionId));
    }
  }, {
    key: "create",
    value: function create(productId, data) {
      return this.client.post("/products/".concat(productId, "/options"), data);
    }
  }, {
    key: "update",
    value: function update(productId, optionId, data) {
      return this.client.put("/products/".concat(productId, "/options/").concat(optionId), data);
    }
  }, {
    key: "delete",
    value: function _delete(productId, optionId) {
      return this.client.delete("/products/".concat(productId, "/options/").concat(optionId));
    }
  }]);

  return ProductOptions;
}();

exports.default = ProductOptions;
