// cezerin2-client/lib/api/products/optionValues.js
"use strict";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

Object.defineProperty(exports, "__esModule", {
  value: true
});

var ProductOptionValues = /*#__PURE__*/function () {
  function ProductOptionValues(client) {
    _classCallCheck(this, ProductOptionValues);

    this.client = client;
  }

  _createClass(ProductOptionValues, [{
    key: "list",
    value: function list(productId, optionId) {
      return this.client.get("/products/".concat(productId, "/options/").concat(optionId, "/values"));
    }
  }, {
    key: "retrieve",
    value: function retrieve(productId, optionId, valueId) {
      return this.client.get("/products/".concat(productId, "/options/").concat(optionId, "/values/").concat(valueId));
    }
  }, {
    key: "create",
    value: function create(productId, optionId, data) {
      return this.client.post("/products/".concat(productId, "/options/").concat(optionId, "/values"), data);
    }
  }, {
    key: "update",
    value: function update(productId, optionId, valueId, data) {
      return this.client.put("/products/".concat(productId, "/options/").concat(optionId, "/values/").concat(valueId), data);
    }
  }, {
    key: "delete",
    value: function _delete(productId, optionId, valueId) {
      return this.client.delete("/products/".concat(productId, "/options/").concat(optionId, "/values/").concat(valueId));
    }
  }]);

  return ProductOptionValues;
}();

exports.default = ProductOptionValues;
