// cezerin2-client/lib/api/products/products.js
"use strict";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

Object.defineProperty(exports, "__esModule", {
  value: true
});

var Products = /*#__PURE__*/function () {
  function Products(client) {
    _classCallCheck(this, Products);

    this.client = client;
    this.resourceUrl = "/products";
  }

  _createClass(Products, [{
    key: "list",
    value: function list(filter) {
      return this.client.get(this.resourceUrl, filter);
    }
  }, {
    key: "retrieve",
    value: function retrieve(id, filter) {
      return this.client.get("".concat(this.resourceUrl, "/").concat(id), filter);
    }
  }, {
    key: "create",
    value: function create(data) {
      return this.client.post(this.resourceUrl, data);
    }
  }, {
    key: "update",
    value: function update(id, data) {
      return this.client.put("".concat(this.resourceUrl, "/").concat(id), data);
    }
  }, {
    key: "delete",
    value: function _delete(id) {
      return this.client.delete("".concat(this.resourceUrl, "/").concat(id));
    }
  }, {
    key: "skuExists",
    value: function skuExists(productId, sku) {
      return this.client.get("".concat(this.resourceUrl, "/").concat(productId, "/sku"), {
        sku: sku
      });
    }
  }, {
    key: "slugExists",
    value: function slugExists(productId, slug) {
      return this.client.get("".concat(this.resourceUrl, "/").concat(productId, "/slug"), {
        slug: slug
      });
    }
  }]);

  return Products;
}();

exports.default = Products;
