// cezerin2-client/lib/api/products/variants.js
"use strict";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

Object.defineProperty(exports, "__esModule", {
  value: true
});

var ProductVariants = /*#__PURE__*/function () {
  function ProductVariants(client) {
    _classCallCheck(this, ProductVariants);

    this.client = client;
  }

  _createClass(ProductVariants, [{
    key: "list",
    value: function list(productId) {
      return this.client.get("/products/".concat(productId, "/variants"));
    }
  }, {
    key: "create",
    value: function create(productId, data) {
      return this.client.post("/products/".concat(productId, "/variants"), data);
    }
  }, {
    key: "update",
    value: function update(productId, variantId, data) {
      return this.client.put("/products/".concat(productId, "/variants/").concat(variantId), data);
    }
  }, {
    key: "delete",
    value: function _delete(productId, variantId) {
      return this.client.delete("/products/".concat(productId, "/variants/").concat(variantId));
    }
  }, {
    key: "setOption",
    value: function setOption(productId, variantId, data) {
      return this.client.put("/products/".concat(productId, "/variants/").concat(variantId, "/options"), data);
    }
  }]);

  return ProductVariants;
}();

exports.default = ProductVariants;
